stock_minimum_shelf_life/models/stock_move.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2019 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from datetime import datetime
+
+from odoo.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT as dt_fmt
+from odoo import models
+
+
+class StockQuant(models.Model):
+    _inherit = 'stock.quant'
+
+    def _gather(self, product_id, location_id, lot_id=None, package_id=None,
+                owner_id=None, strict=False):
+        res = super()._gather(
+            product_id, location_id, lot_id=lot_id,
+            package_id=package_id, owner_id=owner_id, strict=strict
+        )
+
+        # filter the quants that respect the minimum shelf life date
+        if "minimum_expiry_shelf_life" in self.env.context:
+            minimum_expiry_shelf_life = (
+                self.env.context.get('minimum_expiry_shelf_life')
+            )
+            quants = self.search(
+                [('id', 'in', res.ids),
+                 ('lot_id.removal_date', '>', minimum_expiry_shelf_life)
+                 ]
+            )
+            return quants
+
+        return res
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    def _action_assign(self):
+        proc_sales_moves = self.filtered(
+            lambda m: m.picking_id.group_id.sale_id
```

Comment:
@jgrandguillaume are you sure that this should be limited to stock moves for sales to customers? What about checking it also in production orders, so that you don't accept to consume products that have a certain shelf life limit? This is very important, for example, for food processing.

Al alternative could be that the user defines in settings when they want to check minimum shelf life:
1) Shipping to customers (destination location in the stock move is type customer)
2) Consuming products for production (destination location in the stock move is type production)

Reply:
@jbeficent I agree with you, this is a very good remark. 

Will take this into account.

Reply:
Not so easy because if in pick pack ship setup, then the pick make the reservation.. so it's not a customer one yet..